CODE-GUIDELINES.md

```diff
@@ -327,386 +110,290 @@ Follow this comment structure:
+Always them to define `margin` and `padding`
 
 ```sass
-// Bad
-.header .search-box input[type=search] + .button {}
-
-.pf-search-box {
-  .btn {}
+.nav-link {
+  padding-bottom: $pf-spacer-xxxs;
 }
 
-
-// Good
-.pf-search__button {}
+.dropdown-menu {
+  margin-left: $pf-spacer-sm;
+}
 ```
 
-### Nesting
-
-Avoid Sass nesting, but if you must do it follow the [inception rule](http://thesassway.com/beginner/the-inception-rule) and never go more than three layers deep.
-
-Limit nesting to the following use cases:
+You can also use [Bootstrap spacing utilities](http://v4-alpha.getbootstrap.com/utilities/spacing/) with PatternFly sizes:
 
-1. Media queries
-1. Parent selectors
-1. States, pseudo-classes and pseudo-elements
-1. Overwrite Bootstrap
+```html
+<p class="mb-lg">This paragraph has a large bottom margin</p>
+```
 
+##### Comment all magic numbers
 
-For longer style blocks don't nest the modifier code as it reduced the legibility of the code.
+If a situation arises where a value needs entering into the authoring style sheets that isn't already defined in the variables this should serve as a red flag to you.
 
-#### 2. Media queries
+In the case where a 'magic' number needs entering in the authoring style sheets, ensure a comment is added on the line above to explain its relevance.
 
-Component-specific media queries should be nested inside the component block. Use [Bootstrap responsive breakpoints mixins](http://v4-alpha.getbootstrap.com/layout/overview/#responsive-breakpoints) and remember that PatternFly is mobile first.
 
-**Do progressive enhancement, not gracefully degrade.**
+#### Shadows
 
+PatternFly has 6 types of shadows:
 
 ```sass
-.pf-nav {
-  ...
-
-  // Styles for small view ports and up
-  @include media-breakpoint-up(sm) { ... }
-}
+$pf-box-shadow-sm
+$pf-box-shadow-md
+$pf-box-shadow-lg
+$pf-box-shadow-inset
+$pf-box-shadow-left
+$pf-box-shadow-right
 ```
 
-#### 3. Parent selectors
-
-Make use of [Sass’s parent selector](https://css-tricks.com/the-sass-ampersand/ mechanism. This allows all rules for a given component to be maintained in one location.
+Always use variables to apply `box-shadow`:
 
 ```sass
-.pf-primary-nav {
-  ...
-
-  // Nav appearing in header: Right-align navigation when it appears in the header
-  .pf-header & {
-    margin-left: auto;
-  }
+.btn {
+  box-shadow: $pf-box-shadow-sm;
 }
 ```
 
-All styles for `.pf-primary-nav` should be found in one place, rather than scattered throughout multiple partial files.
+You can also use box shadow utilities to add or remove shadows to any element:
 
-#### 4. States, pseudo-classes and pseudo-elements
+```html
+<p class="pf-box-shadow-md">This paragraph has a medium box shadow</p>
 
-States of a component should be included as a nested element. This includes hover, focus, and active states:
+<button class="btn btn-primary pf-box-shadow-none">This button has no shadow</button>
+```
 
-```sass
-.btn {
-    background: blue;
+#### Gradients
```

Comment:
should we spell out buttons here?

Reply:
I'll leave this open till we have the new designs